src/main/java/io/github/azagniotov/metrics/reporter/cloudwatch/CloudWatchReporter.java

```diff
@@ -278,16 +278,16 @@ private void processTimer(final String metricName, final Timer timer, final List
         if (builder.withZeroValuesSubmission || snapshot.size() > 0) {
             for (final Percentile percentile : builder.percentiles) {
                 final double convertedDuration = convertDuration(snapshot.getValue(percentile.getQuantile()));
-                stageMetricDatum(true, metricName, convertedDuration, durationUnit, percentile.getDesc(), metricData);
+                stageMetricDatum(true, metricName, convertedDuration, rateUnit, percentile.getDesc(), metricData);
             }
         }
 
         // prevent empty snapshot from causing InvalidParameterValueException
         if (snapshot.size() > 0) {
             final String formattedDuration = String.format(" [in-%s]", getDurationUnit());
-            stageMetricDatum(builder.withArithmeticMean, metricName, convertDuration(snapshot.getMean()), durationUnit, DIMENSION_SNAPSHOT_MEAN + formattedDuration, metricData);
-            stageMetricDatum(builder.withStdDev, metricName, convertDuration(snapshot.getStdDev()), durationUnit, DIMENSION_SNAPSHOT_STD_DEV + formattedDuration, metricData);
-            stageMetricDatumWithConvertedSnapshot(builder.withStatisticSet, metricName, snapshot, durationUnit, metricData);
+            stageMetricDatum(builder.withArithmeticMean, metricName, convertDuration(snapshot.getMean()), rateUnit, DIMENSION_SNAPSHOT_MEAN + formattedDuration, metricData);
```

Comment:
Should not these changes be conditional? e.g.: only send rateUnit instead of `durationUnit` if some flag has been set?

Reply:
Thats definitely a valid Point. Haven't thought about that before.
Changed this in second Commit